joshprzybyszewski/cribbage
# Using DynamoDB as an option for persistence

## What broke / What you're adding
We have an in-memory DB, mysql DB, and mongodb DB. Unfornuately, in-memory is ephemeral, mysql is stood up in RDS and we pay for uptime, and I don't like mongodb. So let's add a DB that is pay-as-you-use and can store our games for us in aws.

## How you did it
Add another set of services to talk to dynamodb using the AWS go sdk v2 (v1 is deprecated or something).
Also, stand up a local dynamodb instance in docker-compose so that you can test locally.

## How to test it and how to try to break it
1. Run `make dockerbuild`
2. In the docker-compose, change the image of `cribbage-server` to `image: cribbage:latest`.
3. go to `localhost:18080` and start playing a game
4. tear down the dynamodb container (using ctop) and notice that your games are all borked.
